### PR TITLE
BaSP-MR-130: [Trainer] Move post and put logic

### PR DESCRIPTION
**Pull request by ghost**
**Replace the data from the local state
**Move the logic of the post to the thunk file in the activities folder and apply it to the entity index file.
**Move the logic of the put to the thunk file in the activities folder and apply it to the entity index file.
**To make the user experience more pleasant, for each request, a spinner (isLoading) should be implemented so that the user understands that a request is being made, for example when loading the table.

### CODE REVIEW

**Review by Federico-Peralta — APPROVED** [review]
Good job!

**Comment by Federico-Peralta** [comment]
[[Trainers] - Move post and put logic](https://trello.com/c/KiKANroj/130-trainers-move-post-and-put-logic)

**Review by ale-schmidt — CHANGES_REQUESTED** [review]
I think you could take out the "**dispatch(getTrainers());**" function inside the "**handleDeleteConfirmation**" function:

```
const handleDeleteConfirmation = async () => {
    if (selectedTrainerId) {
      try {
        await dispatch(deleteTrainer(selectedTrainerId));
        setModalMessage('Trainer deleted successfully');
        setIsConfirmationModal(false);
        setTypeStyle('success');
        dispatch(getTrainers());
      } catch (error) {
        console.error('Error deleting trainer:', error);
        setModalMessage('Error deleting trainer');
        setTypeStyle('error');
        setStateModal('fail');
      }
```

File: src/Components/Trainers/Form/index.jsx [hunk 1]
```diff
@@ -3,88 +3,60 @@ import { useHistory, useParams } from 'react-router-dom';
...
     }
-  }, []);
-
-  const handleCloseModal = () => {
-    setShowModal(false);
-    if (shouldRedirect) {
-      history.push('/trainers');
-    }
-  };
+  }, [id]);
```
Comment (ale-schmidt) [inline src/Components/Trainers/Form/index.jsx]: I think you can dismiss this Id param inside the useEffect function

**Review by ale-schmidt — CHANGES_REQUESTED** [review]

File: src/Components/Trainers/Form/index.jsx (hunk 1 above)
Comment (ghost) [reply src/Components/Trainers/Form/index.jsx]: I prefer to eliminate it from the fas of the earth

**Review by ale-schmidt — APPROVED** [review]
all good!

**Comment by LucreciaV** [comment]
great!

**Pull request merged**